llvm/test/Feature/Repo/repo_pruning_dependents.ll

```diff
@@ -0,0 +1,63 @@
+; This test checks the program repository pruning pass for the dependents.
+;
+; In the llvm-project-prepo project, the GO's digest could be generated either by the
+; RepoMetadataGenerator pass or by the RepoObjectWriter. When a new global object (GO)
+; is generated by an LLVM optimisation pass after the RepoMetadataGenerator pass, the
+; digest will be generated by the RepoObjectWriter. If a fragment has an external fixup
+; referencing this GO, we add the GO's compilation member address to the dependents.
+;
+; When a fragment is pruned, all its dependents need to be recorded in the repo.tickets
+; metadata to guarantee that the dependents (those GO generated by later optimizations)
+; are present in the final compilation record.
+;
+; In the test, the digest of 'A' is given in the TicketNode (as if it were generated
+; by the RepoMetadataGenerator pass). The digests of `B` and `C` will be generated by the
+; RepoObjectWriter. Since Function 'A' is dependent on function 'B' and 'B' is dependent
+; on function 'C', both TicketNodes of 'B' and 'C' need to be added into the 'repo.tickets'
+; during the pruning.
+;
+; The test includes eight steps:
```

Comment:
Saying how many steps in advance adds another thing to update, and I don't think the number of steps is important to what is being tested. Maybe this could be something more generic like:

`; The steps in this test are:`

Reply:
Fixed, thanks.